scriptmodules/emulators/vice/vice.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+BIN="${0%/*}/$1"
+ROM="$2"
+
+rootdir="/opt/retropie"
+romdir="${ROM%/*}"
+ext="${ROM##*.}"
+source "$rootdir/lib/archivefuncs.sh"
+
+archiveExtract $ROM "crt,d64,g64,prg,t64,tap,x64,vsf"
+
+# check successful extraction and if we have at least one file
+if [[ $? == 0 && ${#arch_files[@]} -gt 0 ]]; then
+    ROM=${arch_files[0]}
+    romdir=$arch_dir
```

Comment:
This launch script might be small enough to just embed in the module btw - and that way you can avoid hardcoding "rootdir". You have to quote some of the additional vars, but it might be worth it.

you could return 2 for empty archive to save the check here and do it in the archive function ?